module2/projects/coupon-codes-api/index.md

```diff
@@ -0,0 +1,467 @@
+
+Updates a coupon from inactive to active
+
+```bash
+PATCH /api/v1/merchants/:merchant_id/coupons/:id
+Content-Type: application/json
+Accept: application/json
+
+{
+  "status": "active"
+}
+```
+
+```json
+status: 200
+body:
+
+{
+  "data": {
+    "id": "1",
+    "type": "coupon",
+    "attributes": {
+      "name": "Buy One Get One 50",
+      "code": "BOGO50",
+      "amount_off": 50,
+      "percent_off": null,
+      "status": "active",
+      "merchant_id": 1,
+      "uses": 2
+    }
+  }
+}
+```
+</section>
+
+----
+
+## 2) Iterating on Existing Endpoints
+
+<section class="dropdown">
+### 6. Merchant Coupon Index Sorted
+
+When passed a query param, returns coupons sorted by active or inactive
+
+**Request**
+```bash
+GET /api/v1/merchants/:merchant_id/coupons?sort=status
+Content-Type: application/json
+Accept: application/json
+```
+
+**Response**
+
+```json
+{
+  "data": [
+    {
+      "id": "1",
+      "type": "coupon",
+      "attributes": {
+        "name": "Buy One Get One 50",
+        "code": "BOGO50",
+        "amount_off": 50,
+        "percent_off": null,
+        "status": "active",
+        "merchant_id": 1,
+        "uses": 1
+      }
+    },
+    {
+      "id": "3",
+      "type": "coupon",
+      "attributes": {
+        "name": "One More Coupon",
+        "code": "MONEYOFF",
+        "amount_off": 10,
+        "percent_off": null,
+        "status": "active",
+        "merchant_id": 1,
+        "uses": 2
+      }
+    },
+    {
+      "id": "2",
+      "type": "coupon",
+      "attributes": {
+        "name": "25 Percent Sale",
+        "code": "25PERCENT",
+        "amount_off": null,
+        "percent_off": 25,
+        "status": "inactive",
+        "merchant_id": 1,
+        "uses": 0
+      }
+    },
+  ]
+}
+```
+</section>
+
+<section class="dropdown">
+### 7. Merchant Invoice
+
+Return a merchant's invoice and include the id of the coupon used (if one was used)
+
+**Request**
+```bash
+GET /api/v1/merchants/:merchant_id/invoices/:id
+Content-Type: application/json
+Accept: application/json
+```
+
+**Response**
+
+```json
+{
+  "data": [
+    {
+      "id": "86",
+        "type": "invoice",
+        "attributes": {
+          "customer_id": "17",
+          "merchant_id": "3",
+          "coupon_id": "1",
+          "status": "shipped"
+        }
+    },
+    {
+      "id": "186",
+        "type": "invoice",
+        "attributes": {
+          "customer_id": "39",
+          "merchant_id": "3",
+          "coupon_id": "2",
+          "status": "shipped"
+        }
+    },
+    {
+      "id": "318",
+        "type": "invoice",
+        "attributes": {
+          "customer_id": "67",
+          "merchant_id": "3",
+          "coupon_id": null,
+          "status": "shipped"
+        }
+    }
+  ]
+}
+```
+
+</section>
+
+<section class="dropdown">
+### 8. Merchants
+
+Update the merchants endpoint to include a count of coupons for each merchant and a count of invoices with coupons applied for each merchant.
```

Comment:
Love these endpoints that require extending existing endpoints!

Reply:
Same!

Quick question, for task 7 ad maybe even 8, have we removed the requirements for showing the subtotal (total w/o coupons) and grand total (total w/ coupons)?

Reply:
I think we have because that would involve more complex AR than we'll expect them to know. Does that sound right @juliet-e ?